src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -88,11 +90,10 @@ public void save( final URI identifier, final T element ) throws IOException {
 		final Path path = Paths.get( identifier );
 		final File file = path.toFile();
 		FileUtil.ensureFolder( path.toFile() );
-		try ( final Output output = new Output( new LZ4FrameOutputStream( newOutputStream( path ) ) ) ) {
-			output.writeString( version );
+		try ( OutputStream outputStream = newOutputStream( path ) ) {
```

Comment:
Please use try-with-resources here, just pass in the OutputStream (similarly to `load`). You should always close the outer most stream only, which should take care of dependent streams.
```
try {
    return save( newOutputStream( path ), element );
} catch ( final Throwable t ) {
    ....
}
```

Reply:
Hmm. I am sorry. Maybe I do not understand correctly. But this currently is try-with-resource, right?

Reply:
Closing a stream multiple times is no problem. Closing an outer stream will also close an inner stream, but closing an inner stream does not close the outer stream. Therefore, close twice ... just to make sure.